minicluster/src/main/java/org/apache/accumulo/miniclusterImpl/MiniAccumuloClusterImpl.java

```diff
@@ -1003,31 +1002,23 @@ public synchronized void stop() throws IOException, InterruptedException {
     } catch (RuntimeException e) {
       log.error("Error zapping zookeeper locks", e);
     }
-    control.stop(ServerType.ZOOKEEPER, null);
 
     // Clear the location of the servers in ZooCache.
-    // When ZooKeeper was stopped in the previous method call,
-    // the local ZooKeeper watcher did not fire. If MAC is
-    // restarted, then ZooKeeper will start on the same port with
-    // the same data, but no Watchers will fire.
-    boolean startCalled = true;
+    // If MAC is restarted, then ZooKeeper will start
+    // on the same port with the same data, but no Watchers will fire.
```

Comment:
I don't think that's true. The watchers should see the disconnected event when  MAC is stopped, and either a syncconnected or an expired session, depending on how long it is offline when it restarts and whether that time exceeds the connect timeout.

The real reason for clearing the locations is that the ephemeral nodes for the locks corresponding to the stopped Accumulo processes will still exist, because the ZK session still exists, but since these services were restarted, they need to create new locks.

So, it's not about the watchers not firing... it's just that the locks are no longer up-to-date with the currently running Accumulo services.

Reply:
removed this comment in a481f9b